src/overview.md

```diff
@@ -0,0 +1,173 @@
+high-quality production experiences across a wide variety of compilation targets
+(web, mobile, desktop).
+
+Dart also forms the foundation of [Flutter](https://flutter.dev): it provides
+the language and runtimes that power Flutter apps, but also supports many core
+developer tasks like formatting, analyzing, and testing code.
+
+## Dart: The Language
+
+The Dart language is type safe: it uses static type checking to ensure that a
+variable’s value always matches the variable’s static type, sometimes referred
+to as sound typing. Although types are mandatory, type annotations are optional
+because of type inference. The Dart typing system is also flexible, allowing the
+use of a 'dynamic' type combined with runtime checks, which can be useful during
+experimentation or for code that needs to be especially dynamic.
+
+Dart offers [sound null safety](https://dart.dev/null-safety), meaning that
+values can’t be null unless you say they can be. With sound null safety, Dart
+can protect you from null exceptions at runtime through static code analysis.
+Unlike many other null-safe languages, when Dart determines that a variable is
+non-nullable, that variable is always non-nullable: if you inspect your running
+code in the debugger, you’ll see that non-nullability is retained at runtime
+(hence "sound" null safety).
+
+This code sample showcases a number of Dart language features: libraries, async
+calls, nullable and non-nullable types, arrow syntax, generators, streams and
+getters. A more comprehensive set of annotated language examples can be found in
+[Dart by Example](http://jpryan.me/dartbyexample/) by
+[@jryanio](https://twitter.com/jryanio).
+
+```dart
+import 'dart:async';
+import 'dart:math' show Random;
+
+main() async {
+  print('Compute π using the Monte Carlo method.');
+  await for (final estimate in computePi().take(100)) {
+    print('π ≅ $estimate');
+  }
+}
+/// Generates a stream of increasingly accurate estimates of π.
+Stream<double> computePi({int batch: 100000}) async* {
+  var total = 0;
+  var count = 0;
+  while (true) {
+    final points = generateRandom().take(batch);
+    final inside = points.where((p) => p.isInsideUnitCircle);
+
+    total += batch;
+    count += inside.length;
+    final ratio = count / total;
+
+    // Area of a circle is A = π⋅r², therefore π = A/r².
+    // So, when given random points with x ∈ <0,1>,
+    // y ∈ <0,1>, the ratio of those inside a unit circle
+    // should approach π / 4. Therefore, the value of π
+    // should be:
+    yield ratio * 4;
+  }
+}
+Iterable<Point> generateRandom([int? seed]) sync* {
+  final random = Random(seed);
+  while (true) {
+    yield Point(random.nextDouble(), random.nextDouble());
+  }
+}
+class Point {
+  final double x, y;
+  const Point(this.x, this.y);
+  bool get isInsideUnitCircle => x * x + y * y <= 1;
+}
+```
+
+[This sample can be run
+live](https://nullsafety.dartpad.dev/b2e3a914bd39d0408d0580c42ef2e58b) using
```

Comment:
This can be https://dartpad.dev/b2e3a914bd39d0408d0580c42ef2e58b?null_safety=true